source/extensions/retry/priority/other_priority/other_priority.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+
+#include "common/upstream/load_balancer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Priority {
+
+class OtherPriorityRetryPriority : public Upstream::RetryPriority {
+public:
+  OtherPriorityRetryPriority(uint32_t update_frequency) : update_frequency_(update_frequency) {}
+  ~OtherPriorityRetryPriority() {
+    if (callback_) {
+      callback_->remove();
+    }
+  }
+
+  const Upstream::PriorityLoad&
+  determinePriorityLoad(const Upstream::PrioritySet& priority_set,
```

Comment:
Thinking about this some more I don't think this is actually safe, since the set is owned by the LB which is owned by the cluster. If a cluster is removed during routing, we'd run into issues accessing the set. I'll try to figure out a better way to keep the priority load in sync with membership updates, or perhaps not handle that at all and document the limitation.

Reply:
Changed it to no longer persist the priority_set address and instead documented the limitation. While watching cluster membership is possible it gets tricky due to the possibility of cluster being removed, so I'd rather tackle that separately to adding this basic implementation